src/request_handler.cpp

```diff
@@ -90,7 +92,14 @@ bool RequestHandler::is_host_up(const Address& address) const {
   return io_worker_->is_host_up(address);
 }
 
+void RequestHandler::start_request() {
+  start_time_ns_ = uv_hrtime();
+}
+
 void RequestHandler::set_response(Response* response) {
+  uint64_t elapsed = uv_hrtime() - start_time_ns_;
+  connection_->metrics()->requests.record_value(elapsed / 1000);
+  connection_->metrics()->request_rates.mark();
```

Comment:
Just a thought: would it make sense to to ``metrics()->record_request(<latency>)" and have the metrics class update the latency histo and mark the meter?